pkg/performance/registry.go

```diff
@@ -19,15 +39,102 @@ var registry = make(map[MetricType]NewContinuousCollector)
 // This function is usually called during package initialization (typically in init() functions)
 // to register collector implementations before they can be instantiated by performance.Manager.
 //
-// It will panic if a collector for the given metricType is already registered
+// On non-Linux platforms, this is a no-op to allow unit tests to run on macOS/Windows.
+// It will panic if a collector for the given metricType is already registered on Linux.
 func Register(metricType MetricType, collector NewContinuousCollector) {
+	// No-op on non-Linux platforms
+	if runtime.GOOS != "linux" {
+		registryLogger.V(1).Info("Skipping collector registration on non-Linux platform",
+			"metric_type", metricType, "platform", runtime.GOOS)
+		return
+	}
+
```

Comment:
I think we can have a simpler API and just have one Register method: What `Register` would do is:
- Check the capability requirements for the collector
- If the running platform can't support the collector just log an error saying we're missing capabilities.

In combination with my other comment for non linux platforms this would always log missing capabilities if attempting to register  linux collectors